src/Ztm.WebApi.Tests/Controllers/IssueTokenControllerTests.cs

```diff
@@ -0,0 +1,297 @@
+            );
+
+            Assert.Throws<ArgumentNullException>(
+                "watcher",
+                () => new IssueTokenController(this.factory.Object, this.configuration, null, this.callbackRepository.Object, this.ruleRepository.Object)
+            );
+
+            Assert.Throws<ArgumentNullException>(
+                "callbackRepository",
+                () => new IssueTokenController(this.factory.Object, this.configuration, this.watcher.Object, null, this.ruleRepository.Object)
+            );
+
+            Assert.Throws<ArgumentNullException>(
+                "ruleRepository",
+                () => new IssueTokenController(this.factory.Object, this.configuration, this.watcher.Object, this.callbackRepository.Object, null)
+            );
+        }
+
+        [Fact]
+        public async Task PostAsync_WithValidArgumentAndUrlWasNotSet_ShouldSuccessButNotAddCallback()
+        {
+            // Arrange.
+            var amount = PropertyAmount.One;
+            var note = "Test Issuing";
+            var tx = NBitcoin.Transaction.Parse(TestTransaction.Raw1, ZcoinNetworks.Instance.Mainnet);
+            var fee = Money.Satoshis(500);
+
+            this.propertyManagementRpc.Setup
+            (
+                c => c.GrantAsync
+                (
+                    It.Is<Property>(p => p.Id == this.zcoinConfiguration.Property.Id && p.Type == this.zcoinConfiguration.Property.Type),
+                    this.zcoinConfiguration.Property.Issuer.Address,
+                    this.zcoinConfiguration.Property.Distributor.Address,
+                    It.Is<PropertyAmount>(a => a.Equals(amount)),
+                    note,
+                    It.IsAny<CancellationToken>()
+                )
+            ).ReturnsAsync(tx).Verifiable();
+
+            this.rawTransactionRpc.Setup
+            (
+                c => c.SendAsync
+                (
+                    tx,
+                    It.IsAny<CancellationToken>()
+                )
+            ).ReturnsAsync(tx.GetHash()).Verifiable();
+
+            var req = new IssueTokenRequest
+            {
+                Amount = amount,
+                Note = note,
+            };
+
+            var httpContext = new DefaultHttpContext();
+            this.subject.ControllerContext = new ControllerContext
+            {
+                HttpContext = httpContext
+            };
+
+            // Act.
+            var result = await this.subject.PostAsync(req, CancellationToken.None);
+
+            // Assert.
+            this.propertyManagementRpc.Verify();
+            this.rawTransactionRpc.Verify();
+
+            var okObjectResult = Assert.IsType<OkObjectResult>(result);
+            var returnedTx = okObjectResult.Value;
+            var properties = returnedTx.GetType().GetProperties();
+            var txVal = (uint256)properties.First(p => p.Name == "Tx").GetValue(returnedTx);
+
+            Assert.Equal(tx.GetHash(), txVal);
```

Comment:
Better to use [FluentAssertion](https://fluentassertions.com/objectgraphs/) on this so we don't need to use reflection like this.